src/autotype/AutoType.cpp

```diff
@@ -506,80 +534,64 @@ QList<AutoTypeAction*> AutoType::createActionFromTemplate(const QString& tmpl, c
     return list;
 }
 
-QString AutoType::autoTypeSequence(const Entry* entry, const QString& windowTitle)
+/**
+ * Retrive the autotype sequences matches for a given windowTitle
+ */
+QList<QString> AutoType::autoTypeSequences(const Entry* entry, const QString& windowTitle)
 {
+    QList<QString> sequenceList;
+
     if (!entry->autoTypeEnabled()) {
-        return QString();
+        return sequenceList;
     }
 
-    bool enableSet = false;
-    QString sequence;
     if (!windowTitle.isEmpty()) {
-        bool match = false;
         const QList<AutoTypeAssociations::Association> assocList = entry->autoTypeAssociations()->getAll();
         for (const AutoTypeAssociations::Association& assoc : assocList) {
             const QString window = entry->resolveMultiplePlaceholders(assoc.window);
             if (windowMatches(windowTitle, window)) {
                 if (!assoc.sequence.isEmpty()) {
-                    sequence = assoc.sequence;
+                    sequenceList.append(assoc.sequence);
                 } else {
-                    sequence = entry->defaultAutoTypeSequence();
+                    sequenceList.append(entry->effectiveAutoTypeSequence());
                 }
-                match = true;
-                break;
             }
         }
 
-        if (!match && config()->get("AutoTypeEntryTitleMatch").toBool() &&
+        if (config()->get("AutoTypeEntryTitleMatch").toBool() &&
             windowMatchesTitle(windowTitle, entry->resolvePlaceholder(entry->title()))) {
-            sequence = entry->defaultAutoTypeSequence();
-            match = true;
+            sequenceList.append(entry->effectiveAutoTypeSequence());
         }
 
-        if (!match && config()->get("AutoTypeEntryURLMatch").toBool() &&
+        if (config()->get("AutoTypeEntryURLMatch").toBool() &&
             windowMatchesUrl(windowTitle, entry->resolvePlaceholder(entry->url()))) {
-            sequence = entry->defaultAutoTypeSequence();
-            match = true;
+            sequenceList.append(entry->effectiveAutoTypeSequence());
         }
 
-        if (!match) {
-            return QString();
+        if (sequenceList.isEmpty()) {
+            return sequenceList;
         }
     } else {
-        sequence = entry->defaultAutoTypeSequence();
+        sequenceList.append(entry->effectiveAutoTypeSequence()); 
     }
 
     const Group* group = entry->group();
     do {
-        if (!enableSet) {
-            if (group->autoTypeEnabled() == Group::Disable) {
-                return QString();
-            } else if (group->autoTypeEnabled() == Group::Enable) {
-                enableSet = true;
-            }
+        if (group->autoTypeEnabled() == Group::Disable) {
```

Comment:
Maybe you should do this check before filling `sequenceList` with Auto-Type associations and then throwing them away.